design_temp/clean.py

```diff
@@ -0,0 +1,147 @@
+import csv
+import re
+from dataclasses import dataclass
+
+
+# This script generates design_temp_by_county.csv
+# from the raw text copied from the ENERGY STAR pdf located at:
+# https://www.energystar.gov/ia/partners/bldrs_lenders_raters/downloads/County%20Level%20Design%20Temperature%20Reference%20Guide%20-%202015-06-24.pdf
+
+
+def remove_junk(content):
+    table_title = (
+        "Exhibit 1: Cooling and Heating Design Temperatures by State and County"
+    )
+    content = content.split(table_title)[1]
+    pattern = r"[ ]+ENERGY STAR Certified Homes\nCounty-Level Design Temperature Reference Guide"
+    content, subs = re.subn(pattern, "", content)
+    if subs != 115:
+        raise Exception
+    pattern = (
+        r"State[\s]+County[\s]+1% Cooling Temperature \(°F\)[\s]+"
+        r"99% Heating Temperature \(°F\)[\s]+HDD/CDD Ratio[\s]+"
+        r"Weather Station Selected for Cooling Temperature[\s]+"
+        r"Reference[\s]+Weather Station Selected for Heating Temperature[\s]+Reference"
+    )
+    content, subs = re.subn(pattern, "", content)
+    if subs != 47:
+        raise Exception
+    pattern = (
+        r"1% Cooling State County Temperature\n\(°F\)\n"
+        r"99% Heating HDD/CDD Weather Station Selected for "
+        r"Weather Station Selected for\nTemperature Ratio Cooling "
+        r"Temperature Reference Heating Temperature Reference \(°F\)"
+    )
+    content, subs = re.subn(pattern, "", content)
+    if subs != 69:
+        raise Exception
+    pattern = r"[ ]+Page [0-9]+ of 116"
+    content, subs = re.subn(pattern, "", content)
+    if subs != 116:
+        raise Exception("subs: {}".format(subs))
```

Comment:
Minor nitpick, but is there a reason why this exception has an argument but not the others?

Reply:
No particular reason, I'll add the argument to the other excepts